agent/testagent.go

```diff
@@ -239,15 +246,15 @@ func (a *TestAgent) Start() (err error) {
 
 // waitForUp waits for leader election, or waits for the agent HTTP
 // endpoint to start responding, depending on the agent config.
-func (a *TestAgent) waitForUp() error {
+func (a *TestAgent) waitForUp(name string) error {
```

Comment:
Could we pull this out of `a.Agent.config.NodeName` or maybe store it on the TestAgent?

Reply:
In general I think arguments to a function are much more desirable than struct fields accessed from methods on the same struct. Function arguments have much smaller scope and are easier to follow.

But I agree this is a weird argument to be accepting on a function called `waitForUp`. Since this value is only used in the error message I think I can remove it and have the caller add it by wrapping the error.

Good call out!